src/components/classifier/SwipeCard.js

```diff
@@ -19,24 +31,57 @@ class SwipeCard extends Component {
             overlayViewHeight: subjectDisplayHeight,
             overlayAnswerIndex: 0,
             imageIsVisible: false,
+            localUri: ''
         }
+        this.unlinkImageOnLoad = false
+
+        this.updateOverlayImageForImageDimensions = this.updateOverlayImageForImageDimensions.bind(this)
     }
 
     componentDidMount() {
-        this.props.panX.addListener((value) => {
+        this.listenerId = this.props.panX.addListener((value) => {
+                this.setState({ overlayAnswerIndex: value.value < 0 ? 0 : 1})
+        })
+
+        const remoteUri = this.props.subject.display.src
+        loadRemoteImageToCache(remoteUri).then((localUri) => {
+            if (this.unlinkImageOnLoad) {
+                RNFetchBlob.fs.unlink(localUri)
+                return
+            }
+
+            const nativeImage = new NativeImage(localUri)
+            nativeImage.getImageSize().then(this.updateOverlayImageForImageDimensions)
+
+            this.props.imageActions.saveImageLocation(remoteUri, localUri)
             this.setState({
-                overlayAnswerIndex: value.value < 0 ? 0 : 1
+                localUri: localUri
```

Comment:
This can be `this.setState({ localUri })`, but I wouldn't worry about it unless there's anything else that I find needs adjusting here.

Reply:
I'll update